class Solution {
    public int searchInsert(int[] nums, int target) {
        int low=0;
        int high=nums.length-1;
        int ans=0;
        int mid;
        while(low<=high)
        {
            mid=(low+high)/2;

            if(nums[mid]==target)
            {
                return mid;
            }
            if(nums[mid]<target)
            {
                ans=mid+1;
                low=mid+1;
            }
            else
            {
                ans=mid;
                high=mid-1;
            }
        }
        return ans;
    }
}